packages/synapse-react-client/src/components/SynapseTable/SearchV2.tsx

```diff
@@ -164,14 +164,25 @@ class _Search extends React.Component<InternalSearchProps, SearchState> {
           }
         additionalFilters.push(columnMultiValueQueryFilter)
       } else {
-        const columnSingleValueQueryFilter: ColumnSingleValueQueryFilter = {
-          columnName,
-          operator: ColumnSingleValueFilterOperator.LIKE,
-          values: [`%${searchText}%`],
-          concreteType:
-            'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+        if (columnModel?.columnType == ColumnTypeEnum.ENTITYID) {
+          const columnSingleValueQueryFilter: ColumnSingleValueQueryFilter = {
+            columnName,
+            operator: ColumnSingleValueFilterOperator.EQUAL,
+            values: [`${searchText.toLowerCase().replace('syn', '')}`],
```

Comment:
Let's do an exact match for an ID  (remove the "syn", and set the operator to EQUAL)